ghcide/src/Development/IDE/Core/Shake.hs

```diff
@@ -535,6 +535,7 @@ data IdeState = IdeState
     ,shakeExtras          :: ShakeExtras
     ,shakeDatabaseProfile :: ShakeDatabase -> IO (Maybe FilePath)
     ,stopMonitoring       :: IO ()
+    ,rootDir              :: FilePath
```

Comment:
This needs some more documentation. What does this `rootDir` point to? Is it the root of the cradle we found, the argument provided by `--cwd DIR`.

If we want to follow this approach to the extreme, we need a `Note` that explains our high level strategy how we avoid the global variable `$CWD` (queries and modified by `getCurrentDirectory` and `setCurrentDirectory`)

Reply:
done